Test.cs

```diff
@@ -12,7 +14,12 @@ public static long calculate(long number)
                 throw new Exception("Number must be non-negative.");
             }
 
```

Comment:
The recursive call to `Calculate` should decrement the `number` parameter to avoid infinite recursion. It should be `number * Calculate(number - 1)`.